### ISSUE TITLE
MeshAgent show as connected but no more Desktop nor Terminal TAB

### ISSUE THREAD

**Issue by tootai** [Ylianst/MeshCentral#2700]
Hi all,
I face a problem with a Windows10 20H2 netbook: both server (Debian10) and agent are on the same network. MeshAgent is started manually by user when he/she need support. What happend is that I see stealthily the agent disappearing in my device list and in events log "session terminated ...". On the agent desktop nothing wrong, the MeshAgent windows still up and disconnect button is available.

On the server side, the agent is shown as connected but the 2 tabs Desktop and Terminal are no more present. The console one is still here -I can connect-, sending/receiving messages is working too. Problem is I can't no more connect to the desktop nor cmd line. I have other agent which are running automaticaly in background at computer start who doesn't show this behavior.

Is there a way to bring back both tabs even using a cmd in console mode ?

Thanks for your fabulous job, great software.

**Comment by VINISHVV** [comment]
Please try any of the below method once.

1) go to agent console and run 'agentupdate'
2)go to console -> click on agent action
      Perform clear the core
      Perform upload default server core.

**Comment by tootai** [comment]
> agentupdate
Self Update Failed because this agent is not an instance of (Mesh Agent)

What does this mean ?

**Comment by tootai** [comment]
OK, clearing the core & upload default server core did the trick :) Many thanks. Could you explain the above error ?

**Comment by VINISHVV** [comment]
> > agentupdate
> > Self Update Failed because this agent is not an instance of (Mesh Agent)
> 
> What does this mean ?

Can you post the screenshot of this error ?

**Comment by tootai** [comment]
![Capture d’écran de 2021-05-27 19-29-01](https://user-images.githubusercontent.com/6564304/119871116-6e953e80-bf22-11eb-933b-23fbccc9ff30.png)

Thanks

**Comment by Ylianst** [comment]
Ok. The problem here is that the server was updated from a very old version to the latest version. This is a bug Bryan needs to fix. There is the work around: You need to downgrade your server to MeshCentral v0.7.1 for a while and wait for all the agents to update... so, keep that version for a few weeks... then once all agents are updated, switch to the latest version. Steps are:

Go in the folder above "node_modules" and "meshcentral-data". Then, do this:

- Stop the server from running in the background.
- Rename "node_modules" to "node_modules_bak".
- run "npm install meshcentral@0.7.1".
- run "node node_modules/meshcentral"
- Wait for all extra modules to be installed and the server to run.
- Hit Ctrl-C
- Start the server again in the background.

Once v0.7.1 is running, all your agents will update and will work fine. In a few weeks, update your server to latest.

This is a bug Bryan needs to fix.

Hope that helps,
Ylian

**Comment by Ylianst** [comment]
By the way, thank you @VINISHVV. That was exactly the right questions to ask!

**Comment by krayon007** [comment]
> > > agentupdate
> > > Self Update Failed because this agent is not an instance of (Mesh Agent)
> > 
> > 
> > What does this mean ?
> 
> Can you post the screenshot of this error ?

Normally updating the agent, is for when the agent is running as a background service. If the instance of the agent you are attempting to self update, is not running as a background service, you'll get that error response. The name within the paranthesis, is the service name it was checking.

**Comment by tootai** [comment]
So if I understand well it should NOT display this message even while not running in background and you -Bryan- are working on it. If I don't want to see this message I can downgrade my server till problem is corrected. Am I right ?

**Comment by Ylianst** [comment]
If you downgrade MeshCentral to v0.7.1, wait for all agents to update and update the server to latest, it should work. Basically, your using v0.7.1 as an intermediate version. If Bryan fixed it, then going directly to latest from an early version would work.

**Comment by Ylianst** [comment]
Ok, after talking to Bryan... Bryan will need to fix this for sure. If the agent is run in temporary mode, the current code can't update it and the new server will send data that the old agent can't understand. The best is to stay with an older version until Bryan fixes it.

Good news is that we have other internal customers with the same issue, so this is a priority.

**Comment by tootai** [comment]
I removed old node_modules and reinstalled: now I'm running 0.7.99 Is it OK ?

**Comment by Ylianst** [comment]
Yes, if that work for you and the tabs are back, that is ok. Bryan is working on a fix.

**Comment by tootai** [comment]
I tried with a Mac OS client, still get the 2 tabs disappearing. With the above VINISHVV  solution they come back to live.
